client/src/Login.tsx

```diff
@@ -1,63 +1,147 @@
+      console.log("From server:", data)
+      setLoading(false);
+      // The fourth datapoint is whether the kube config was set properly
+      // Only attempt to log the user in if kube config was created
+      if(data[4]) {
+        loginUser(false);
+      } else {
+        alert(badCredentials);
+      }
+    })
+    // Why is this executing twice?
+    // Response after parsing local cred/config files
+    window.electronAPI.onSendConfig('onSendConfig', (event: any, data: [string, string, string, string]) => {
+      console.log("Parsed from local files:", data);
+      setLoading(false);
+      // Update each input field
+      setAccessKey(data[0]);
+      setSecretKey(data[1]);
+      setClusterName(data[2]);
+      setRegionName(data[3]);
+    })
 
-  // If the server sends anything besides 200, then the user isn't authenticated
-  // For the user's convenience, find and parse their three files (in $HOME/.aws/ 
-  // and $HOME/.kube/) and populate their corresponding fields in the form
-  const parseLocalCredentials = () => {
-    console.log('<ELECTRON NEEDED> : Parse User\'s local files here!');
-  }
+    return () => {
+      console.log('useEffect: unmount');
+      window.electronAPI.unMount();
+    };
+  }, [])
 
-  const loginUser = () => {
+  const loginUser = (loadConfig : boolean) => {
     //alert('id:' + accessKey + ' secret:' + secretKey);
 
     fetch('http://localhost:3000/authenticate', {
       method: 'GET'
     })
-    .then(response => {
-      //console.log('resp:', resp.json());
-      //console.log('typeof:', typeof response.json());
-      //console.log(response.json());
-      console.log('status:', response.status);
-      if(response.status !== 200) {
-        console.log('bad credentials...');
+      .then(response => {
+        //console.log('resp:', resp.json());
+        //console.log('typeof:', typeof response.json());
+        //console.log(response.json());
+        console.log('status:', response.status);
+        if (response.status !== 200) {
+          console.log('bad credentials...');
+          alert(badCredentials)
+          // Leverage Electron to parse authentication data from local files
+          if(loadConfig) window.electronAPI.getConfig();
+          return;
+        }
+        else {
+          console.log('Valid creds!');
+          navigate('/app');
+          return;
+        }
+      }).catch(err => {
+        alert('Unable to log in. Please ensure the server is running.')
         // Leverage Electron to parse authentication data from local files
-        parseLocalCredentials();
-      }
-      else {
-        console.log('Valid creds!');
-        navigate('/app');
-      }
-    })
+        if(loadConfig) window.electronAPI.getConfig();
+        return;       
+      })
   }
 
-  const buttonPressed = (event : FormEvent) => {
-    event.preventDefault();
+  // Returns true if the input string is empty
```

Comment:
This comment is not needed. A useful comment here though could be explaining why the trim is necessary. Are some string being assigned whitespaces even though they are not empty? It could be very helpful to developers to know that; which I assume is the case since you're running string.trim() in here.